### ISSUE TITLE
Request to reconciler to delete cluster is ignored if the cluster is currently being reconciled

### ISSUE THREAD

**Issue by varbanv** [kyma-incubator/reconciler#480]
**Description**
If KEB sends a request to the mothership to delete a cluster, the request is ignored if that cluster is currently in reconciliation cycle.

**Expected result**
Cluster is deleted.

**Actual result**
Cluster continues regular reconciliation cycles.

**Steps to reproduce**
Delete cluster while it's currently running a reconciliation cycle.

**Troubleshooting**

Here is the relevant log from the mothership:
```
{"level":"INFO","time":"2021-11-25T19:49:21.695Z","message":"Inventory watcher added runtime 'ff4078b3-2413-417e-baa7-287dad4c2af1' to scheduling queue (clusterVersion:416/configVersion:418/status:delete_pending)"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.695Z","message":"Begin transactional DB context"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.708Z","message":"Starting reconciliation for cluster 'ff4078b3-2413-417e-baa7-287dad4c2af1': set cluster status to 'reconciling'"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.708Z","message":"Begin transactional DB context"}
{"level":"INFO","time":"2021-11-25T19:49:21.824Z","message":"ReconRepo found existing reconciliation for cluster 'ff4078b3-2413-417e-baa7-287dad4c2af1' (configVersion:418) which was created at '2021-11-25 19:47:21.826777 +0000 +0000': cannot create another one"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.824Z","message":"Rollback transactional DB context"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.825Z","message":"Cancelling reconciliation for cluster 'ff4078b3-2413-417e-baa7-287dad4c2af1': cluster is already enqueued"}
{"level":"DEBUG","time":"2021-11-25T19:49:21.825Z","message":"Rollback transactional DB context"}
{"level":"ERROR","time":"2021-11-25T19:49:21.826Z","message":"cluster 'ff4078b3-2413-417e-baa7-287dad4c2af1' is already considered for reconciliation (schedulingID:ff4078b3-2413-417e-baa7-287dad4c2af1--a2b83d4c-6c94-4134-a9fa-7342c4c2a822)"}
{"level":"DEBUG","time":"2021-11-25T19:49:35.706Z","message":"empty X-Envoy-External-Address header"}
{"level":"DEBUG","time":"2021-11-25T19:49:39.375Z","message":"empty X-Envoy-External-Address header"}
{"level":"DEBUG","time":"2021-11-25T19:49:39.376Z","message":"Begin transactional DB context"}
{"level":"DEBUG","time":"2021-11-25T19:49:39.383Z","message":"Commit transactional DB context"}
{"level":"DEBUG","time":"2021-11-25T19:49:51.124Z","message":"Bookkeeper evaluated reconciliation (schedulingID:ff4078b3-2413-417e-baa7-287dad4c2af1--a2b83d4c-6c94-4134-a9fa-7342c4c2a822) for cluster 'ff4078b3-2413-417e-baa7-287dad4c2af1' to cluster status 'ready': Done=CRDs,cluster-essentials,istio-configuration,certificates,monitoring,eventing,ory,api-gateway,service-catalog-addons,cluster-users,rafter,application-connector,logging,service-ca
```

**Comment by tobiscr** [comment]
We have to ensure that the finalization of a reconciliation is only overriding the cluster state if it is `reconcling`. Any other cluster states have to stay and should not be updated!

**Comment by tobiscr** [comment]
fix provided in #494 + #499
